astroquery/noirlab/core.py

```diff
@@ -90,12 +102,138 @@ def query_region(self, coordinate, radius=0.1, cache=True):
+            The target region which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.1 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used.
+
+        Returns
+        -------
+        response : `requests.Response`
+        """
+        self._validate_version()
+
+        ra, dec = coordinate.to_string('decimal').split()
+        url = f'{self.siaurl}?POS={ra},{dec}&SIZE={radius}&format=json'
+        response = self._request('GET', url,
+                                 timeout=self.TIMEOUT,
+                                 cache=cache)
+        response.raise_for_status()
+        return response
+
+    def core_fields(self, cache=True):
+        """List the available CORE fields. CORE fields are faster to search
+        than AUX fields.."""
+        response = self._request('GET', self._adsc_url,
+                                 timeout=self.TIMEOUT,
+                                 cache=cache)
+        response.raise_for_status()
+        return response.json()
+
+    def aux_fields(self, instrument, proctype, cache=True):
+        """List the available AUX fields. AUX fields are ANY fields in the
+        Archive FITS files that are not core DB fields.  These are generally
+        common to a single Instrument, Proctype combination. AUX fields are
+        slower to search than CORE fields. """
+        url = f'{self._adsa_url}/{instrument}/{proctype}/'
+        response = self._request('GET', url, timeout=self.TIMEOUT, cache=cache)
+        response.raise_for_status()
+        return response.json()
+
+    def categoricals(self, cache=True):
+        """List the currently acceptable values for each 'categorical field'
+        associated with Archive files.  A 'categorical field' is one in
+        which the values are restricted to a specific set.  The specific
+        set may grow over time, but not often. The categorical fields are:
+        collection, instrument, obs_mode, proc_type, prod_type, site, survey,
+        telescope.
+        """
+        url = f'{self._adsurl}/cat_lists/?format=json'
+        response = self._request('GET', url, timeout=self.TIMEOUT, cache=cache)
+        response.raise_for_status()
+        return response.json()
+
+    @class_or_instance
+    def query_metadata(self, qspec, limit=1000, cache=True):
+        self._validate_version()
+        url = f'{self._adss_url}/?limit={limit}'
+
+        if qspec is None:
+            jdata = {"outfields": ["md5sum", ], "search": []}
+        else:
+            jdata = qspec
+
+        response = self._request('POST', url, json=jdata, timeout=self.TIMEOUT)
+        response.raise_for_status()
+        return astropy.table.Table(rows=response.json())
```

Comment:
it might be better to split this onto several lines to enable easier debugging when (inevitably, in my experience) either the table parser or the json itself breaks somehow

Reply:
I'm producing the json response on web-service side.  If that breaks my web-service tests will break.  Debugging such there is easier than here anyhow. If astropy changes the behavor of the table constructor, I'll have much bigger problems than this.  No change.